src/standalone/Snow/snow_parameterizations.jl

```diff
@@ -49,10 +54,206 @@ end
+        nothing,
+        10,
+    )
+    T_s = sol.root
+    return T_s
+end
+
+function calc_U(ρ_l::FT, d::FT, c::FT, T::FT, T0::FT)::FT where {FT} # assuming q_l = 1
+    return ρ_l * d * c * (T - T0)
+end
+
+function snow_surface_temperature(
+    u_air::FT,
+    T_air::FT,
+    P_air::FT,
+    z_0m::FT,
+    z_0b::FT,
+    q_air::FT,
+    h_air::FT,
+    SWE::FT,
+    T_bulk::FT,
+    ρ_sfc::FT,
+    energy_runoff::FT,
+    LW_d::FT,
+    SW_d::FT,
+    parameters,
+) where {FT}
+    earth_param_set = parameters.earth_param_set
+    _LH_v0::FT = LP.LH_v0(earth_param_set)
+    _ρ_liq::FT = LP.ρ_cloud_liq(earth_param_set)
+    ρ_snow::FT = parameters.ρ_snow
+    h_sfc::FT = snow_depth(SWE, ρ_snow, _ρ_liq)
+    d = FT(0.2)
+    if h_sfc < d
+        return (T_s = T_bulk, ΔF = FT(0))
+    end
+    thermo_params = LP.thermodynamic_parameters(earth_param_set)
+    thermal_state_air =
+        Thermodynamics.PhaseEquil_pTq(thermo_params, P_air, T_air, q_air)
+    cp_m::FT = Thermodynamics.cp_m(thermo_params, thermal_state_air)
+    ϵ_sfc::FT = parameters.ϵ_snow
+    α_sfc::FT = parameters.α_snow
+    d_sfc = FT(0)
+
+    state_air = SurfaceFluxes.StateValues(
+        h_air - d_sfc - h_sfc,
+        SVector{2, FT}(u_air, 0),
+        thermal_state_air,
+    )
+    surface_flux_params = LP.surface_fluxes_parameters(earth_param_set)
+    function conditions_func(T_s)
+        update_conditions(
+            T_s,
+            thermo_params,
+            ρ_sfc,
+            state_air,
+            z_0m,
+            z_0b,
+            surface_flux_params,
+        )
+    end
+    E0(T_s) = conditions_func(T_s)[1]
+    r_ae(T_s) = conditions_func(T_s)[2]
+    LH(T_s) = _LH_v0 * E0(T_s)
+
+    ρ_air = Thermodynamics.air_density(thermo_params, thermal_state_air)
+    SH(T_s) = cp_m * -ρ_air * (T_air - T_s) / r_ae(T_s)
+
+    _σ = LP.Stefan(earth_param_set)
+    F_R(T_s) = -(1 - α_sfc) * SW_d - ϵ_sfc * (LW_d - _σ * T_s^4)
+
+    F_h(T_s) = SH(T_s) + LH(T_s) + F_R(T_s) + energy_runoff
+    # κ = parameters.κ_ice
+    κ = snow_thermal_conductivity(ρ_snow, parameters)
+    f(T_s) = κ * (T_s - T_bulk) / (h_sfc / 2) + F_h(T_s)
+
+    # derivatives for Newton's method
+    SH_prime(T_s) = cp_m * ρ_air / r_ae(T_s)
+
+    ∂LHF∂qc(T_s) = ρ_air * _LH_v0 / r_ae(T_s)
+    ∂qc∂T(T_s) = partial_q_sat_partial_T_ice(P_air, T_s)
+    LH_prime(T_s) = ∂LHF∂qc(T_s) * ∂qc∂T(T_s)
+
+    F_R_prime(T_s) = 4 * ϵ_sfc * _σ * T_s^3
+
+    F_h_prime(T_s) = SH_prime(T_s) + LH_prime(T_s) + F_R_prime(T_s)
+    f_prime(T_s) = (2 * κ / h_sfc) + F_h_prime(T_s)
+
+    fns(T_s) = [f(T_s), f_prime(T_s)]
+    T_s::FT = solve_Ts(fns, T_bulk)
+
+    # adjust T_s and internal energy accordingly if T_s > freezing temperature
+    if T_s > 273.15
+        T_s_new = FT(273.15)
+        _T0::FT = FT(LP.T_0(earth_param_set))
+        U_Ts::FT = calc_U(_ρ_liq, d, cp_m, T_s, _T0)
+        U_Tf::FT = calc_U(_ρ_liq, d, cp_m, T_s_new, _T0)
+        ΔU::FT = U_Ts - U_Tf
+        ΔF::FT = ΔU / parameters.Δt
```

Comment:
Should this be the opposite sign? If T_sfc > T_freeze, energy went into melting the top layer. If we say that instead T_sfc maxed out at T_freeze and the additional energy that would have increased the temperature went into the bulk snow, shouldnt delta F be negative? (negative = towards snow)

I also wonder if we should bring the top layer to q_l of the bulk, not to q_1 = 1

Reply:
On this branch, we have the sign convention where this is correct. After rebasing, we need to change it